modules/text2vec-gpt4all/vectorizer/objects.go

```diff
@@ -0,0 +1,153 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2023 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package vectorizer
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/fatih/camelcase"
+	"github.com/weaviate/weaviate/entities/models"
+	"github.com/weaviate/weaviate/entities/moduletools"
+	"github.com/weaviate/weaviate/modules/text2vec-gpt4all/ent"
+)
+
+type Vectorizer struct {
+	client Client
+}
+
+func New(client Client) *Vectorizer {
+	return &Vectorizer{
+		client: client,
+	}
+}
+
+type Client interface {
+	Vectorize(ctx context.Context, text string) (*ent.VectorizationResult, error)
+}
+
+// IndexCheck returns whether a property of a class should be indexed
+type ClassSettings interface {
```

Comment:
I think this comment belongs somewhere else

Reply:
removed comment